sky/shell/tracing_controller.cc

```diff
@@ -33,7 +108,22 @@ void TracingController::StartTracing() {
   StartBaseTracing();
 }
 
-void TracingController::StopTracing(const base::FilePath& path) {
+void TracingController::StopTracing(const base::FilePath& path,
+                                    bool terminateLoopWhenDone) {
+  auto runner = base::MessageLoop::current()->task_runner();
+  runner->PostTask(FROM_HERE, base::Bind(&TracingController::StopTracingAsync,
+                                         weak_factory_.GetWeakPtr(), path,
```

Comment:
It's scary that this function can be called from a random background thread whereas weak_factory_ is destroyed on the Java/UIKit thread.  I guess there's a lock inside DartVM to synchronize with Dart_RegisterRootServiceRequestCallback?